src/core/algorithms/dc/parser/dc_parser.cpp

```diff
@@ -0,0 +1,168 @@
+#include "algorithms/dc/parser/dc_parser.h"
+
+#include <stdexcept>
+
+#include <boost/algorithm/string.hpp>
+
+#include "algorithms/dc/model/dc.h"
+#include "algorithms/dc/model/operator.h"
+#include "algorithms/dc/model/predicate.h"
+#include "model/table/column_layout_relation_data.h"
+#include "model/table/typed_column_data.h"
+
+namespace mo = model;
+
+namespace {
+
+using namespace algos::dc;
+
+std::unique_ptr<ColumnOperand> GetOperand(ColumnOperand const& operand, std::string str_val) {
+    mo::Type const* type = operand.GetType();
+    return std::make_unique<ColumnOperand>(str_val, type);
+}
```

Comment:
is it correct to take `std::string` here by value?

Reply:
Why do we need uniq pointers here? I see that the only usage is in `DCParser::GetOperands`, which uses it to get uniq pointers, but returns a pair of dereferenced ones aka a pair of `ColumnOperand` object.
Can't we just do
```cpp
ColumnOperand GetOperand(const ColumnOperand& operand, const std::string& str_val) {
    return ColumnOperand(str_val, operand.GetType());
}
```
It will be copy-elided.